docs/3-implementing.md

```diff
@@ -66,3 +73,55 @@ plain field (`foo`) will be of type `keyword`, with no nested field.
 Despite the fact that IDs are often integers in various systems, this is not
 always the case. Since we want to make it possible to map as many data sources
 to ECS as possible, we default to using the `keyword` type for IDs.
+
+## Using non-ECS fields in your event stream
+
+The goal of ECS is to define a common set of fields to make working across various
+streams easier. Each stream will have their own particularities, however, and ECS
+cannot account for all possibilities. It's therefore important that implementers
+feel at ease adding fields and top level objects in their schema that are
+not defined in ECS. This is expected.
+
+Doing so will however incur a risk of conflicting with future versions of ECS.
+Let's review some strategies to reduce the risk of conflicts happening.
+
+### Conflicts
+
+Before going into strategies, let's define what constitutes a conflict.
+
+#### Conflicts with ECS
+
+* A custom field has the same name as a new ECS field, but an incompatible type:
+  * `float` vs `long`
```

Comment:
You list here only a small set of examples with conflicting types. I think it should be made clear that this are some examples and above could be just stated `have a different type`. This is correct in most cases as very few types are compatible.

Reply:
I guess this comment supersedes the other. Yeah I'll word it this way, it will be much simpler. Something along the lines of:

> * A custom field has the same name as a new ECS field, but an incompatible type (most fields are incompatible).

Reply:
Yeah I read your comment too fast. I will reformulate similar to what I wrote here. But I'll also keep the examples, and make it clearer that these are just examples, as you suggest.